suite/volume/volume_setup.go

```diff
@@ -0,0 +1,320 @@
+	return &FluxTest{
+		Test: Test{
+			DefaultSetup:            s,
+			OnChainCheckAttemptsOpt: spec.OnChainCheckAttemptsOpt,
+			Nodes:                   clNodes,
+			Adapter:                 adapter,
+			Prom:                    prom,
+		},
+		FluxInstances:      &fluxInstances,
+		ContractsToJobsMap: contractToJobsMap,
+		NodesByHostPort:    nodesByHostPort,
+		roundsDurationData: make([]float64, 0),
+		AlreadySeenRuns:    make(map[string]bool),
+	}, nil
+}
+
+// deployFluxInstance deploy one flux instance concurrently, add jobs to all the nodes
+func deployFluxInstance(d *FluxInstanceDeployment, g *errgroup.Group) {
+	g.Go(func() error {
+		log.Info().Int("Instance ID", d.Index).Msg("Deploying contracts instance")
+		fluxInstance, err := d.Suite.Deployer.DeployFluxAggregatorContract(d.Suite.Wallets.Default(), d.Spec.FluxOptions)
+		if err != nil {
+			return err
+		}
+		err = fluxInstance.Fund(d.Suite.Wallets.Default(), big.NewInt(0), big.NewInt(1e18))
+		if err != nil {
+			return err
+		}
+		err = fluxInstance.UpdateAvailableFunds(context.Background(), d.Suite.Wallets.Default())
+		if err != nil {
+			return err
+		}
+		// set oracles and submissions
+		err = fluxInstance.SetOracles(d.Suite.Wallets.Default(),
+			contracts.SetOraclesOptions{
+				AddList:            d.Oracles,
+				RemoveList:         []common.Address{},
+				AdminList:          d.Oracles,
+				MinSubmissions:     uint32(d.Spec.RequiredSubmissions),
+				MaxSubmissions:     uint32(d.Spec.RequiredSubmissions),
+				RestartDelayRounds: uint32(d.Spec.RestartDelayRounds),
+			})
+		if err != nil {
+			return err
+		}
+		for _, n := range d.Nodes {
+			fluxSpec := &client.FluxMonitorJobSpec{
+				Name:            fmt.Sprintf("%s_%d", d.Spec.JobPrefix, d.Index),
+				ContractAddress: fluxInstance.Address(),
+				PollTimerPeriod: d.Spec.NodePollTimePeriod,
+				// it's crucial not to skew rounds schedule for that particular volume test
+				IdleTimerDisabled: true,
+				PollTimerDisabled: false,
+				ObservationSource: client.ObservationSourceSpec(d.Adapter.InsideDockerAddr + "/variable"),
+			}
+			job, err := n.CreateJob(fluxSpec)
+			if err != nil {
+				return err
+			}
+			d.FluxInstancesMu.Lock()
+			d.NodesByHostPort[n.URL()] = n
+			d.ContractToJobsMap[fluxInstance.Address()] = append(d.ContractToJobsMap[fluxInstance.Address()],
+				JobByInstance{
+					ID:       job.Data.ID,
+					Instance: n.URL(),
+				})
+			d.FluxInstancesMu.Unlock()
+		}
+		d.FluxInstancesMu.Lock()
+		*d.FluxInstances = append(*d.FluxInstances, fluxInstance)
+		d.FluxInstancesMu.Unlock()
+		return nil
+	})
+}
+
+// roundsStartTimes gets run start time for every contract
+func (vt *FluxTest) roundsStartTimes() (map[string]float64, error) {
+	mu := &sync.Mutex{}
+	g := &errgroup.Group{}
+	contactsStartTimes := make(map[string]float64)
+	for contractAddr, jobs := range vt.ContractsToJobsMap {
+		contractAddr := contractAddr
+		jobs := jobs
+		g.Go(func() error {
```

Comment:
The routine defined in this function should be separated into it's own function, as it'll be useful elsewhere. I think we need an `actions` package that has more complex logic that's re-usable.